tests/test_transform.py

```diff
@@ -37,6 +38,122 @@ def makeRawArrayData(nPoints, nAxes, delta=0.123):
 def makeRawPointData(nAxes, delta=0.123):
     return [i*delta for i in range(nAxes)]
 
+
+def makeEndpoint(name, nAxes):
+    """Make an endpoint
+
+    @param[in] name  one of "Generic", "Point2", "Point3" or "SpherePoint"
+    @param[in] nAxes  number of axes; ignored if the name is not "Generic"
+    """
+    endpointClassName = name + "Endpoint"
+    endpointClass = getattr(afwGeom, endpointClassName)
+    if name == "Generic":
+        return endpointClass(nAxes)
+    return endpointClass()
+
+
+def makeGoodFrame(name, nAxes):
+    """Return the appropriate frame for the given name and nAzes
+
+    @param[in] name  one of "Generic", "Point2", "Point3" or "SpherePoint"
+    @param[in] nAxes  number of axes; ignored if the name is not "Generic"
+    """
+    return makeEndpoint(name, nAxes).makeFrame()
+
+
+def makeBadFrames(name, nAxes):
+    """Return a list of 0 or more frames that are not a valid match for name and nAxes
```

Comment:
Thank you for realizing `nAxes` was irrelevant. The point of this subroutine is to construct a collection of 0 or more astshim frames that are not compatible with the named endpoint. Since all frames are compatible with `GenericEndpoint` and the number of axes is known for all other endpoints, nAxes is indeed irrelevant.